help/develop/admin/deploy/deploy-docker-swarm.md

```diff
@@ -0,0 +1,119 @@
+for example [AWS S3](https://aws.amazon.com/s3/) and Local File System storage.
+
+This document contains instructions to deploy Datagrok using [Docker Swarm](https://docs.docker.com/engine/swarm/)
+on virtual machines or dedicated hosts with Local File System for persistent storage.
+
+More information about Datagrok design and components:
+
+* [Architecture](../architecture.md)
+* [Infrastructure](../infrastructure.md)
+
+## Prerequisites
+
+1. We use native [Docker compose](https://docs.docker.com/compose/) commands to run applications on machines. It
+   simplifies multi-container application development and deployment.
+    1. Download and install the latest version of [Docker Compose](https://docs.docker.com/compose/install/) to your
+       environment (virtual machines, dedicated hosts, etc.)
+
+## Preparations
+
+1. Activate docker swarm mode on your manager node (it should be a datagrok host):
+
+   ```shell
+   docker swarm init
+   ```
+
+   The shell will print a token for adding worker nodes something like that:
+
+   ```shell
+   Swarm initialized: current node (bvz81updecsj6wjz393c09vti) is now a manager.
+
+   To add a worker to this swarm, run the following command:
+
+    docker swarm join \
+    ---token SWMTKN-1-testetsttestetst-qwewsdwqdwdqwddwqdwq XX.XX.XX.XX:2377
+
+   To add a manager to this swarm, run 'docker swarm join-token manager' and follow the instructions.
+   ```
+
+2. Use the received token on worker nodes to join the swarm as described below. You can get
+   token any time by typing in the manager node shell for workers:
+
+   ```shell
+   docker swarm join-token worker
+
+   To add a worker to this swarm, run the following command:
+
+    docker swarm join --token SWMTKN-1-klkjojoijoidowqijd-jdiqjdowqdojwqduudu XX.XX.XX.XX:2377
+   ```
+
+   Or, if you want to add another manager node:
+
+   ```shell
+   docker swarm join-token manager
+
+   To add a manager to this swarm, run the following command:
+
+    docker swarm join --token SWMTKN-1-mkmocmomcoemeo334ad-,xowodoxmwddmoqw33 XX.XX.XX.XX:2377
+   ```
+
+3. From the manager node set labels to all nodes in the swarm cluster.
+
+   1. List all nodes:
+
+      ```shell
+      docker node ls
+      ```
+
+   2. Add labels ```role==datagrok``` and ```role==cvm``` to nodes:
+
+      ```shell
+      docker node update --label-add  role==datagrok <prefer_node>
+      ```
+
+4. Download a Docker Compose YAML
+   file: [link](https://github.com/datagrok-ai/public/blob/master/docker/swarm.docker-compose.yaml)
+
+5. Adjust in the downloaded file `GROK_PARAMETERS` values,
+`X_API_KEY`,`DOCKER_REGISTRY`,`DOCKER_REGISTRY_USER`,
+`DOCKER_REGISTRY_PASS`;`POSTGRES_USER` and `POSTGRES_PASSWORD`must be the same as
+`dbAdminLogin` and `dbAdminPassword` in `GROK_PARAMETERS`.
+
+6. Run the docker stack
+
+   ```shell
+   docker stack deploy -c ./swarm,docker-compose.yaml <STACK_NAME>
```

Comment:
Specify the stack name `datagrok`, it will be fine for most clients, who so not want to think about details